src/NHibernate/Loader/Criteria/CriteriaLoader.cs

```diff
@@ -48,6 +51,8 @@ public class CriteriaLoader : OuterJoinLoader
 			resultTypes = walker.ResultTypes;
 			includeInResultRow = walker.IncludeInResultRow;
 			resultRowLength = ArrayHelper.CountTrue(IncludeInResultRow);
+			cachedProjectedTypes = translator.ProjectedTypes;
```

Comment:
@lnu yes, it just set in walker using the same `translator.HasProjection` condition